keps/2076-kueuectl/README.md

```diff
@@ -0,0 +1,342 @@
+
+--queuing-strategy=strategy                # defaults to BestEffortFIFO
+--namespace-selector=selector              # defaults to {} - all namespaces can use the queue
+--reclaim-within-cohort=policy             # defaults to Never
+--preemption-within-cluster-queue = policy # defaults to Never
+--preemption-when-borrowing = policy       # defaults to Never
+
+–-nominal-quota=rfname1:resource1=value,resource2=value,resource3=value
+–-borrowing-limit=rfname1:resource1=value,resource2=value,resource3=value
+–-lending-limit=rfname1:resource1=value,resource2=value,resource3=value
+```
+
+It is possible to create a ClusterQueue with multiple resource flavors/FlavorQuotas inside
+a ResourceGroup and multiple ResourceGroups covering different sets of resources.
+The command will create the appropriate resource groups with resource flavors 
+in the order they appear in the command line. If two settings have at least 
+one common resource quota specified, they will land in the same ResourceGroup.
+
+Output: 
+A simple confirmation, like in regular kubectl create, 
+`clusterqueue.kueue.x-k8s.io/xxxxx created`
+
+
+### Create LocalQueue
+
+Creates a LocalQueue with the given name pointing to specified ClusterQueue.
+The command validates that the target ClusterQueue exists and 
+its namespace selector matches to LocalQueue's namespace.
+
+Format:
+
+```
+kueuectl create lq|localqueue lqname
+–-namespace=namespace                  # uses context's default namespace if not specified
+–-clusterqueue=cqname
+--skip-validation
+```
+
+Output:  
+A simple confirmation `localqueue.kueue.x-k8s.io/xxxxx created`
+
+### List ClusterQueue
+
+List all ClusterQueues, potentially limiting output to those that are active/inactive and 
+matching the label selector. Format:
+
+```
+kueuectl list cq|clusterqueue(s)
+--active=*|true|false
+--selector=selector                   # label selector
+```
+
+Output columns:
+
+* Name
+* Cohort
+* Pending Workloads
+* Admitted Workloads
+* Status (active or not)
+* Age
+
+### List LocalQueue 
+
+Lists LocalQueues that match the given criteria: point to a specific CQ, 
+being active/inactive, belonging to the specified namespace or matching the label
+selector.
+Format: 
+
+```
+kueuectl list lq|localqueue(s) 
+–-namespace=ns                        # uses context's default namespaces if not specified
+--all-namespaces | -A
+-–clusterqueue=clusterqueue 
+–-active=*|true|false
+--selector=selector                   # label selector
+```
+
+Outputs columns:
+
+* Namespace (if -A is used)
+* Name
+* ClusterQueue
+* Pending Workloads 
+* Admitted Workloads 
+* Status (active or not)
+* Age
+
+### List Workloads
+
+Lists Workloads that match the provided criteria. Format:
+
+```
+kueuectl list workloads 
+--namespace=ns                        # uses context's default namespace if not specified
+--all-namespaces | -A
+--clusterqueue=cq 
+-–localqueue=lq 
+-—only-pending
+—-only-admitted
```

Comment:
Instead of `--condition=Pending|Admitted`, could we provide a field selector for more flexibility?

Reply:
Conditions are multiple fields, though.

Or do you expect some form of jsonpath? I wonder if it could become too complex to write for a common use case such as conditions.

OTOH, do you think we should have other fields?

Reply:
@alculquicondor I was imagined similar to the [`kubectl --field-selector`](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/).

I didn't mean that JSONPath.

Reply:
we can probably make it work for most fields, but conditions will still look weird.